src/App.tsx

```diff
@@ -1,15 +1,63 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
+
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+  const [isInputFocus, setIsInputFocus] = useState(false);
+
+  const { name, born, died } = selectPerson || {};
+
+  const selectedPersonInfo = selectPerson
+    ? `${name} (${born} - ${died})`
+    : 'No selected person';
+
+  const handlePesonClick = (person: Person, namePerson: string) => {
+    setSelectPerson(person);
+    setQuery(namePerson);
+  };
+
+  const applyQuery = useMemo(
+    () => debounce(setAppliedQuery, 300),
+    [],
+  );
+
+  const queryInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
not fixed: 
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/4ec10130-b480-4f26-9edf-f31815ffccb7)

hint: you should change `isInputFocus` value on change immediately and after debounce delay